src/js/components/ConfigurationView.js

```diff
@@ -0,0 +1,242 @@
+
+    let {service, versionID} = this.props;
+    fetchVersion(service, versionID);
+  }
+
+  componentWillReceiveProps({service, versionID}) {
+    super.componentWillReceiveProps(...arguments);
+
+    fetchVersion(service, versionID);
+  }
+
+  getGenralSection({id, cmd, cpus, disk, instances, mem}) {
+    let headerValueMapping = {
+      'ID': id,
+      'CPUs': cpus,
+      'Memory (MiB)': mem,
+      'Disk Space (Mib)': disk,
+      'Instances': instances,
+      'Command': cmd
+    };
+
+    return (
+      <DescriptionList className={sectionClassName}
+        hash={headerValueMapping}
+        headline="General" />
+    );
+  }
+
+  getDockerContainerSection({container}) {
+    if (container == null || container.docker == null) {
+      return null;
+    }
+
+    let {docker} = container;
+
+    let headerValueMapping = {
+      'Force Pull Image': docker.forcePullImage,
+      'Image': docker.image,
+      'Network': docker.network,
+      'Parameters': StringUtil.arrayToCommaSeparatedString(docker.parameters),
+      'Privileged': docker.privileged
+    };
+
+    return (
+      <DescriptionList className={sectionClassName}
+        hash={headerValueMapping}
+        headline="Docker Container" />
+    );
+  }
+
+  getPortDefinitionsSection({portDefinitions}) {
+    if (portDefinitions == null) {
+      return null;
+    }
+
+    let portConfigurations = portDefinitions.map(
+      function (portDefinition, index) {
+        let headline = `Port Definition ${index + 1}`;
+
+        if (portDefinition.name) {
+          headline += ` (${portDefinition.name})`;
+        }
+
+        return (
+          <DescriptionList className="nested-description-list"
+            hash={portDefinition}
+            headline={headline}
+            key={index} />
+        );
+      });
+
+    return (
+      <div className={sectionClassName}>
+        <h5>Port Definitions</h5>
+        {portConfigurations}
+      </div>
+    );
+  }
+
+  getHealthChecksSection({healthChecks}) {
+    if (healthChecks == null) {
+      return null;
+    }
+
+    let healthCheckConfigurations = healthChecks.map(
+      function (healthCheck, index) {
+
+        let headerValueMapping = {
+          'Protocol': healthCheck.protocol,
+          'Grace Period Seconds': healthCheck.gracePeriodSeconds,
+          'Interval Seconds': healthCheck.intervalSeconds,
+          'Timeout Seconds': healthCheck.intervalSeconds,
+          'Max Consecutive Failures': healthCheck.maxConsecutiveFailures
+        };
+
+        switch (healthCheck.protocol) {
+          case 'COMMAND':
+            headerValueMapping = Object.assign(headerValueMapping, {
+              'Command': healthCheck.command.value,
+            });
+            break;
+          case 'HTTP':
+            headerValueMapping = Object.assign(headerValueMapping, {
+              'Path': healthCheck.path,
+              'Port': healthCheck.port,
+              'Port Index': healthCheck.portIndex || 'PORT_INDEX',
```

Comment:
Upsi, that's is actually a bug. I somehow screwed that one. I blame the jetlag... The `PORT_INDEX` is actually the default `healthCheck.portType` value which defines the `portIndex` value type.